queries/c/highlights.scm

```diff
@@ -96,12 +110,14 @@
 
 (field_identifier) @property
 (statement_identifier) @label
-(type_identifier) @type
-(primitive_type) @type
-(sized_type_specifier) @type
 
-((identifier) @type
- (#match? @type "^[A-Z]"))
+[
+(type_identifier)
+(primitive_type)
+(sized_type_specifier)
+ ] @type
+
+(_ type: (_) @type)
```

Comment:
This highlights stuff like `(typeof(foo))` wrong. Either be more specific here (identifier pointer ...) or reset the highlighting for things like typeof.

Reply:
would favor an enumration for things where the highlighting is ok, like `identifier` and `pointer_expression` `type_descriptor`.

Reply:
Looks like `typeof` is considered as a function call by the parser, and thus it is a special case of it.

Reply:
but the argument of typeof will then be highlighted as a type

Reply:
Heh no they wont, because `typeof` is considered as a function.

Reply:
Does not matter. `typeof` is highlighted a function. The arguments as type, because (_) sets the hightlight for the whole region.

Reply:
But why? Do you have a screens hot of that?

The match means :
Any node that has a child tagged `type` should highlight this child as a type.
As `typeof`is a function call, it does not have a `type`-tagged child, and thus nothing is highlighted as a type.

Reply:
![image](https://user-images.githubusercontent.com/7189118/86913742-5ad0c600-c11f-11ea-80d1-ad62f895a336.png)

The tag type means only that it's at the position where the type should go. (_) highlights the whole node as type as type. 

![image](https://user-images.githubusercontent.com/7189118/86913961-bf8c2080-c11f-11ea-8f53-5623ed4d2af2.png)

Reply:
This should be better now !